### ISSUE TITLE
[Feature] supporting composite method

### ISSUE THREAD

**Issue by moolawooda** [pyscf/pyscf#1974]
[Thermochemical composite method](https://www.wikiwand.com/en/Quantum_chemistry_composite_methods#:~:text=Quantum%20chemistry%20composite%20methods%20%28also%20referred%20to%20as,lower%20levels%20of%20theory%20with%20larger%20basis%20sets.) are calculation schemes that combining results from different methods to get higher accuracy.

Will PySCF supports some composite methods like [G4 method](https://pubs.aip.org/aip/jcp/article/126/8/084108/308929/Gaussian-4-theory) or [W4 method](https://pubs.aip.org/aip/jcp/article/125/14/144108/295614/W4-theory-for-computational-thermochemistry-In)?

These methods are typecially useful when computing thermalchemical properties.

**Comment by xis19** [comment]
Composite methods are simple -- they are just combinations of different methods, and will be a very good beginner project for python/theo chem newcomers.

**Comment by moolawooda** [comment]
@xis19 They do are simple. But in my opinion, maybe an encapsulated method is also needed. For example, you don't need to manually achieve b3lyp functional every time you want to use it, although it's also simple either to define a customized `ks.xc` attribute or to do some low-level implementations.

**Comment by tberkel** [comment]
I agree that it would be useful to encapsulate these definitions within
PySCF. Our group has a preliminary implementation of the correlation
consistent composite approximation (ccCA) meant for black-box use within
PySCF. Unfortunately, MP4 isn't available in PySCF, which prevents G4 at
this time.
